client/src/margo_client.c

```diff
@@ -240,9 +280,17 @@ int invoke_client_mount_rpc(void)
     LOGDBG("Got response ret=%" PRIi32, ret);
 
     /* get slice size for write index key/value store */
-    unifyfs_key_slice_range = out.max_recs_per_slice;
+    unifyfs_key_slice_range = out.meta_slice_sz;
     LOGDBG("set unifyfs_key_slice_range=%zu", unifyfs_key_slice_range);
 
+    /* get assigned client id, and verify app_id */
+    unifyfs_client_id = (int) out.client_id;
```

Comment:
I didn't look closely, but in the case the client wants to reattach with a certain id, is there a place where a check is done to verify the returned client id matches the expected id?  Or does this get replaced with a new value?

Reply:
The support for reattach is there with the new attach rpc, but we currently don't expose an API that is suitable for reattach. We used to use the client rank passed to unifyfs_mount() as the client id, but that's no longer viable with the server dynamically assigning client ids. One solution I have thought about is maintaining a map from client-provided rank to server-assigned client id, and then introducing an alternative unifyfs_remount(prefix, rank) that would look up the existing mapping and return the previously assigned client id. In general, proper support for reattach requires a broader discussion with the team to define how we think it should work, and then we can define methods to support that.

Reply:
and to better answer your specific question, currently, the client id is always assigned by the output value from the mount rpc.

Reply:
Thanks for the clarification.  I like the idea of a map or something, from a client "name" to a server assigned id.